### PR TITLE
Added ES6 Alternative Intermediate Code Solution

### PR DESCRIPTION
**Pull request by bmorelli25**
# FreeCodeCamp - Wiki Pull Request Template

Added in a new solution utilizing ES6's new spread operator. This is a new solution that works well and is easy to understand.

### CODE REVIEW

File: Algorithm-Return-Largest-Numbers-In-Arrays.md [hunk 1]
```diff
@@ -92,6 +92,35 @@ function largestOfFour(arr) {
...
+  return result;
+}
+```
+
+:rocket: [Run Code] (https://repl.it/CLjU/51)
+
+### Code Explanation:
+
+- Create an Array to hold our results
+- Utilize a for loop to loop throug each subArray
```
Comment (atjonathan) [inline Algorithm-Return-Largest-Numbers-In-Arrays.md]: ~~throug~~ through

File: Algorithm-Return-Largest-Numbers-In-Arrays.md [hunk 2]
```diff
@@ -92,6 +92,35 @@ function largestOfFour(arr) {
...
+- For each subArray, push the maxium value to our result Array
+- The Math.max() function returns the largest of zero or more numbers. It expets multiple values, and one of the ways we can pass values to this function is through the 'Spread' operator
+- The new ES6 spread operator '...' allows an expression to be expanded in places where multiple inputs are expected. '...arr[i]' inputs the entire subArray of values of 'arr' into the Math.max function.
+
+**Reference:-**
+- [Spread Operator]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator)
+
+
```
Comment (atjonathan) [inline Algorithm-Return-Largest-Numbers-In-Arrays.md]: Remove this line

File: Algorithm-Return-Largest-Numbers-In-Arrays.md [hunk 3]
```diff
@@ -92,6 +92,35 @@ function largestOfFour(arr) {
...
+```
+
+:rocket: [Run Code] (https://repl.it/CLjU/51)
+
+### Code Explanation:
+
+- Create an Array to hold our results
+- Utilize a for loop to loop throug each subArray
+- For each subArray, push the maxium value to our result Array
+- The Math.max() function returns the largest of zero or more numbers. It expets multiple values, and one of the ways we can pass values to this function is through the 'Spread' operator
```
Comment (atjonathan) [inline Algorithm-Return-Largest-Numbers-In-Arrays.md]: ~~It expets multiple values,~~ What does this mean?

**Comment by atjonathan** [comment]
@Rafase282, Do we want another solution or this challenge?

File: Algorithm-Return-Largest-Numbers-In-Arrays.md (hunk 3 above)
Comment (bmorelli25) [inline Algorithm-Return-Largest-Numbers-In-Arrays.md]: Sorry, I should've spell checked. It should say "It _expects_ multiple values"

**Comment by atjonathan** [comment]
### **LGTM :+1:**

**Comment by Rafase282** [comment]
Nope, only 3 solutions. @bmorelli25 @atjonathan 

This needs to be fixed.

**Comment by bmorelli25** [comment]
@atjonathan Do you want to leave it as-is or swap out one of the solutions? Let me know and I'll either close the pull request or make the changes

**Comment by Rafase282** [comment]
We would have to decide in which category it falls then to either replace or merge the solution.

**Comment by Rafase282** [comment]
After taking some time to look at the answer I think we gain nothing from adding it. It uses elements from the current intermediate and advance solutions, call it a hybrid if you will. Nothing wrong with the code but I would say why not use map already? and we have one that already does that. I'm closing this for now but feel free to discuss further if you do not agree. @bmorelli25

**Pull request closed without merging**